cores/esp8266/base64.cpp

```diff
@@ -39,33 +38,20 @@ String base64::encode(const uint8_t * data, size_t length, bool doNewLines) {
     // base64 needs more size then the source data, use cencode.h macros
     size_t size = ((doNewLines ? base64_encode_expected_len(length)
                                : base64_encode_expected_len_nonewlines(length)) + 1);
-    char * buffer = (char *) malloc(size);
-    if(buffer) {
+    String base64("-FAIL-");

```

Comment:
I just realized this local name shadows the class name. It would be wise to change it.

Reply:
If you init like this, you're doing a double work internally: first copy the const char "_FAIL_" in, then change that when it's not used. Please reconsider my previous comment: init with default constructor, then assign "-FAIL-" on failure and return, or keep going. As I said, it also allows NRVO to work, which means no object copies on return.

Reply:
done.